internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+
+	for _, zipFile := range zipReader.File {
+		isManifest, err := filepath.Match("*/manifest.json", filepath.ToSlash(zipFile.Name))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to examine file %s inside local archive file %s", zipFile.Name, archivePath)
+		}
+
+		if isManifest {
+			manifestFile, err := zipFile.Open()
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to open manifest file %s inside local archive file %s", zipFile.Name, archivePath)
+			}
+			defer manifestFile.Close()
+
+			jsonByteArr, err := io.ReadAll(manifestFile)
+			if err != nil {
+				return nil, err
+			}
+
+			pluginManifest := new(manifest)
+			if err := json.Unmarshal(jsonByteArr, &pluginManifest); err != nil {
+				return nil, err
+			}
+
+			return pluginManifest, nil
+		}
+	}
+
+	return nil, errors.Errorf(`failed to find manifest file inside local archive file "%s"`, archivePath)
+}
+
+func getRemoteManifest(owner, name, version string) (*manifest, error) {
+	manifestUrl := fmt.Sprintf("https://api.hub.confluent.io/api/plugins/%s/%s", owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	r, err := http.Get(manifestUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to read manifest file from Confuent Hub")
+	}
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	pluginManifest := new(manifest)
+	if err := json.Unmarshal(body, &pluginManifest); err != nil {
+		return nil, err
+	}
+
+	return pluginManifest, nil
+}
+
+func getPluginDirFromFlag(cmd *cobra.Command) (string, error) {
+	if cmd.Flags().Changed("plugin-directory") {
+		pluginDir, err := cmd.Flags().GetString("plugin-directory")
+		if err != nil {
+			return "", err
+		}
+
+		if pluginDir, err = filepath.Abs(pluginDir); err != nil {
+			return "", err
+		}
+
+		if !utils.DoesPathExist(pluginDir) {
+			return "", errors.Errorf(invalidDirectoryErrorMsg, pluginDir)
+		}
+
+		return pluginDir, nil
+	}
+
+	return "", nil
+}
+
+func existingPluginInstallation(pluginDir string, pluginManifest *manifest) ([]string, error) {
+	// Bundled installations
+	if utils.DoesPathExist(filepath.Join(pluginDir, pluginManifest.Name)) {
+		return nil, errors.New("unable to install plugin because it is already bundled")
+	}
+
+	// Other previous installations
+	immediateDirectory := filepath.Join(pluginDir, fmt.Sprintf("%s-%s", pluginManifest.Owner.Username, pluginManifest.Name))
+	uberJar := filepath.Join(pluginDir, fmt.Sprintf("%s-%s.jar", pluginManifest.Name, pluginManifest.Version))
+
+	var installations []string
+	if utils.DoesPathExist(immediateDirectory) {
+		installations = append(installations, immediateDirectory)
+	}
+	if utils.DoesPathExist(uberJar) {
+		installations = append(installations, uberJar)
+	}
+
+	return installations, nil
+}
+
+func uninstall(pathToPlugin string, dryRun, force bool) error {
```

Comment:
Should we add `confluent connect plugin uninstall`?

Reply:
Good idea.

A dedicated uninstall command probably needs a bit more work, though. This function only removes the plugin, but doesn't update the worker files. I'll need to think about what the right way to do that is (since it adds the parent directory and not the plugin directory itself), and probably add this in when I add the on-prem version of `list`.

Reply:
Wait, how does this function get called currently if there's no `uninstall` command?

Reply:
It's called on line 118. The command first checks if this plugin already exists in the plugin directory, and then calls this function if so to prompt the user to confirm removal of the existing version.

Reply:
Oh, it replaces it if it already exists.

Reply:
I'll rename this function to `replaceInstallation` until I get to writing the `uninstall` command.